src/Application/Actions/Campaigns/Get.php

```diff
@@ -38,6 +44,12 @@ protected function action(Request $request, Response $response, array $args): Re
 
         $sfId = $args['salesforceId'] ?? throw new HttpNotFoundException($request);
 
+        $campaignFromMatchbotDB = $this->campaignRepository->findOneBySalesforceId(Salesforce18Id::ofCampaign($sfId));
+
+        if ($campaignFromMatchbotDB) {
```

Comment:
the ticket I'm working on right now is only concerned with the true branch here, but of course we can't break the site in the false branch so then we keep the existing behaviour by proxying from SF.